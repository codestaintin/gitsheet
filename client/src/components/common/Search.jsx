import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

const Search = ({ handleSearch }) => (
  <div className="container">
    <div className="row">
      <div className="col-12">
        <div className="input-group">
          <input
            className="form-control border-secondary py-2"
            type="search"
            placeholder="Search for a git command"
            name="searchTerm"
            onChange={handleSearch}
          />
          <div className="input-group-append">
            <button className="btn btn-outline-dark" type="button">
              <i className="fa fa-search" />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
);

Search.propTypes = {
  handleSearch: PropTypes.func
};

export default Search;