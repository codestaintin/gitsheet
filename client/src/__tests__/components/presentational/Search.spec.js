import React from 'react';
import { shallow } from 'enzyme';
import Search from '../../../components/common/Search';

describe('<Search>', () => {
  const props = {
    handleSearch: jest.fn(() => {})
  };
  it('should render the Search', () => {
    const tree = shallow(<Search {...props} />);
    expect(tree).toMatchSnapshot();
  });
});