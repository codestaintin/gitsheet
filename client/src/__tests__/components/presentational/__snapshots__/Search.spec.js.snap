// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> should render the Search 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <div
        className="input-group"
      >
        <input
          className="form-control border-secondary py-2"
          name="searchTerm"
          onChange={[MockFunction]}
          placeholder="Search for a git command"
          type="search"
        />
        <div
          className="input-group-append"
        >
          <button
            className="btn btn-outline-dark"
            type="button"
          >
            <i
              className="fa fa-search"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
